modules/claims_api/app/controllers/concerns/claims_api/poa_verification.rb

```diff
@@ -81,11 +81,20 @@ def verify_representative_and_veteran(logged_in_representative_user, target_vete
       end
 
       def poa_code_in_organization?(poa_code)
+        organization = ::Veteran::Service::Organization.find_by(poa: poa_code)
         representative = ::Veteran::Service::Representative.where('? = ANY(poa_codes)', poa_code).first
```

Comment:
I'm not sure if this is completely what we want here, at least according to the M21-1 manual. I think what we might actually want to check is this: **if the current user is a representative, verify that _they_ have a POA code for the submitted organization.**

### (A Lot More) Context

Under Section [I.i.2.E.1.e. What Constitutes a Complete VA Form 21-22 or VA Form 21-22a](https://www.knowva.ebenefits.va.gov/system/templates/selfservice/va_ssnew/help/customer/locale/en-US/portal/554400000001018/content/554400000181478/M21-1,-Part-I,-Subpart-i,-Chapter-2,-Section-E---Systems-Updates-for-Power-of-Attorney-(POA)-Appointments#1e), it lists the following as "Required Information:"

> 21-22: signature of the representative acting on behalf of the service organization with a corresponding date of signature.

However, it _also_ says this:

> Important: Completion of Blocks 16A and 16B of VA Form 21-22 is optional. If completed, there is no need to verify that the specific representative listed in Block 16A is accredited. This form documents the appointment of the entire service organization listed in Block 15 and does not limit action to a specific representative within the organization.

The field for block 16A is this:

>NAME OF OFFICIAL REPRESENTATIVE ACTING ON BEHALF OF THE
ORGANIZATION NAMED IN ITEM 15 (This is an appointment of the entire organization
and does not indicate the designation of only this specific individual to act on behalf of the
organization)

Block 16B that person's job title.

Reply:
i.e., I think this might be more along the lines of what we want to be checking instead:

```ruby
def poa_code_in_organization?(poa_code)
    organization = ::Veteran::Service::Organization.find_by(poa: poa_code)

    if user_is_target_veteran?
      organization.present?
    else
      representative = ::Veteran::Service::Representative.find_by(first_name: @current_user.first_name,
                                                                  last_name: @current_user.last_name)
      organization.present? && representative.poa_codes.include?(poa_code)
    end
end
```